pkg/user/tx_client_test.go

```diff
@@ -281,3 +313,35 @@ func (suite *TxClientTestSuite) queryCurrentBalance(t *testing.T) int64 {
 	require.NoError(t, err)
 	return balanceResp.Balances.AmountOf(app.BondDenom).Int64()
 }
+
+func wasRemovedFromTxTracker(txHash string, txClient *user.TxClient) bool {
+	nonce, signer, exists := txClient.GetTxFromTxTracker(txHash)
+	return !exists && nonce == 0 && signer == ""
+}
+
+// asserts that a tx was indexed in the tx tracker and that the sequence does not increase
+func assertTxInTxTracker(t *testing.T, txClient *user.TxClient, txHash string, expectedSigner string) {
+	nonce, signer, exists := txClient.GetTxFromTxTracker(txHash)
+	require.True(t, exists)
+	require.Equal(t, expectedSigner, signer)
+	seq := txClient.Signer().Account(expectedSigner).Sequence()
+	// Successfully broadcast transaction increases the nonce
+	// txInfo is indexed before the nonce is increased
+	require.Equal(t, seq, nonce+1)
+}
```

Comment:
What do you mean by asserting that it doesn't increase yet expecting the sequence to be 1 ahead of the sequence of the transaction (i.e. it was incremented)?